core/lib/eth_client/src/lib.rs

```diff
@@ -154,6 +154,23 @@ pub trait EthInterface: Sync + Send {
     async fn block(&self, block_id: BlockId) -> EnrichedClientResult<Option<Block<H256>>>;
 }
 
+/// Information about the base fees provided by the L1 client.
+#[derive(Debug, Clone, PartialEq, Eq, Default)]
+pub struct L2Fees {
+    pub base_fee_per_gas: u64,
+    pub pubdata_price: U256,
+}
+
+#[async_trait]
+pub trait ZkSyncInterface: Sync + Send {
```

Comment:
Maybe a dumb suggestion: What about providing a single unified interface both for L1 and L2 settlement layers, which would return `base_fee_per_gas` / `base_fee_per_blob_gas` (zero for L2?) / `pubdata_price` (zero for L1)? In this way, some branching code could be simplified.

Reply:
I thought that we wanted to have additional type safety, but if it is acceptable to merge those two, then okay

Reply:
It is not very easy unfortunately, an L2 client also has the same interface as the L1 one, so it is not possible to say that "all clients that only support L1 have this trait implementation, but those that also support L2 have another trait implementation". If keep different traits, but the same return data it starts looking rather weird that we have two similar-looking methods that return the same struct, but apparently with different field values